### DISCUSSION TITLE
New data indicator for the table / data grid component

### DISCUSSION THREAD

**Discussion by sang-yeo** [twilio-labs/paste#960] [Q&A] [answered]
Hello there, here is a user story... As a debugger user, I'd like to identify which error messages are new and old. The current debugger doesn't have/provide an indicator for new errors therefore I keep checking the errors I checked few days ago. 

Here are my questions.... 
- Can I put an indicator? (Please see the initial mock)  
- Which icon should I use? (The initial mock uses same dot as new UDL debugger pulsing dot(?)
<img width="977" alt="new-error-indicator" src="https://user-images.githubusercontent.com/67388401/99747711-ef651080-2a8f-11eb-8ddc-991caa13cad3.png">

**Comment by sang-yeo** [comment]
Here are two more concepts relate to the story above. 
<img width="969" alt="Screen Shot 2020-11-20 at 9 13 17 AM" src="https://user-images.githubusercontent.com/67388401/99829254-ea947100-2b10-11eb-9bc5-274b26a74176.png"> This concept I use a similar treatment as the Gmail which grey background for viewed events and bold text for new errors. 
<img width="983" alt="Screen Shot 2020-11-20 at 9 13 05 AM" src="https://user-images.githubusercontent.com/67388401/99829387-1879b580-2b11-11eb-927b-7ac1b264c174.png">
This concept I use different color for the error icon to differentiate new and viewed error events

**Comment by kate-porter** [comment, marked as answer]
Hey @sang-yeo -

Since this is a table of error logs, it almost doesn't feel necessary to include an error icon at all; that may be redundant, unless there are things in this table that aren't errors.

I like the option that's inspired Gmail, but you wouldn't be able to accomplish the gray background treatment with the Table component. One thing you could try is similar to iMessage, where new errors get a dot, like so:

![image](https://user-images.githubusercontent.com/5374045/99848718-820cba80-2b37-11eb-9712-9eb630e0b641.png)

You could have the new items pulse just once as they come into view, and then keep the static dot there after the single pulse.

The dot is using our `new` token, which we would recommend using here to indicate new errors.

**Reply by sang-yeo** [reply]
Hey @katieporter thanks for the quick response! Yeah, there are two icons, error and warning. That's the reason we try to keep the icon there. But I really like to get rid of it. :) 

What is the meaning of the purple color in the Paste? New console uses a pulsing orange dot to indicate an error, can I use that dot instead? (Which i used on the 1st concept) 
Does the dot pulsing? 

Thanks!!

**Reply by kate-porter** [reply]
Paste uses purple to indicate `new`, which is why I used that in my mock. I'd recommend using that over red. Red generally indicates an error, but this table will be full of errors, and what you are trying to indicate to the user is that some of them are new.

The dot can pulse if you'd like, or it can just be a static circle, like you'd see on unread messages in something like iMessage.

If you'd prefer to use one of the options you shared above, I'd recommend the first. The second option isn't doable with the Table component, and the third option only uses color as an indicator, which should generally be avoided whenever possible.

**Reply by sang-yeo** [reply]
Great! Thank you
